src/stream.cpp

```diff
@@ -75,18 +75,39 @@ int32_t SteamAudioStreamPlayback::_mix(AudioFrame *buffer, double rate_scale, in
 		ls->bufs.in.data[1][i] = mixed_frames[i].y;
 	}
 
+	IPLAirAbsorptionModel airAbsorptionModel{};
+	airAbsorptionModel.type = IPL_AIRABSORPTIONTYPE_DEFAULT;
+
+	IPLVector3 sourcePosition = ipl_vec3_from(ls->src.player->get_global_transform().origin);
+	IPLVector3 listenerPosition = gs->listener_coords.origin;
+	float airAbsorption[3];
+	airAbsorption[0] = ls->cfg.air_absorption_low;
+    airAbsorption[1] = ls->cfg.air_absorption_mid;
+    airAbsorption[2] = ls->cfg.air_absorption_high;
+
+	iplAirAbsorptionCalculate(gs->ctx, sourcePosition, listenerPosition, &airAbsorptionModel, airAbsorption);
```

Comment:
Could we put all of this inside `if (ls->cfg.is_air_absorp_on)`? There's not much need for computing all of this if the feature is disabled.